core/api/src/domain/authentication/index.types.d.ts

```diff
@@ -24,21 +24,39 @@ type IdentityPhone = IdentityBase & {
   phone: PhoneNumber
   email: undefined
   emailVerified: undefined
+
+  username?: undefined
```

Comment:
maybe I am missing something but kratos is not updated when the user sets the username.. so do we have a process syncing this?

Reply:
This is different. It's added to `IdentityPhone`/`IdentityEmail`/`IdentityPhoneEmail` just to satisfy the `AnyIdentity` interface and it really only has meaning in `IdentityDeviceAccount` where the type is `IdentityUsername` and not our user's account usernames